app/src/main/java/com/example/sharingang/ReportFragment.kt

```diff
@@ -0,0 +1,107 @@
+package com.example.sharingang
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.RadioButton
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.navArgs
+import com.example.sharingang.databinding.FragmentReportBinding
+import com.example.sharingang.users.UserRepository
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+
+@AndroidEntryPoint
+class ReportFragment : Fragment() {
+
+    private val args: ReportFragmentArgs by navArgs()
+    private lateinit var binding: FragmentReportBinding
+    private lateinit var reporterId: String
+    private lateinit var reportedId: String
+    private lateinit var reportedUsername: String
+    @Inject
+    lateinit var userRepository: UserRepository
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentReportBinding.inflate(inflater, container, false)
+        reporterId = args.reporterId
+        reportedId = args.reportedId
+        reportedUsername = args.reportedName
+
+
+        val radioButtons = listOf(
+            binding.radioUsername, binding.radioProfilePicture, binding.radioItem, binding.radioOther
+        )
+        setupRadioButtons(radioButtons)
+        setupCancelButton()
+        setupButtonOk(radioButtons)
+        binding.textReportedUsername.text = "Reporting $reportedUsername"
```

Comment:
I'd add a string in values/strings.xml for that. Also check the warning: "don't concatenate text"